sklearn/ensemble/tests/test_iforest.py

```diff
@@ -200,3 +200,16 @@ def test_max_samples_consistency():
     X = iris.data
     clf = IsolationForest().fit(X)
     assert_equal(clf.max_samples_, clf._max_samples)
+
+
+def test_iforest_subsampled_features():
+    rng = check_random_state(4)
+
+    X_train, X_test, y_train, y_test = train_test_split(iris.data,
+                                                        iris.target,
+                                                        random_state=rng)
+    # Training classifier with sub-sampled features
+    clf = IsolationForest(max_features=0.8)
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict(X_test)
+    assert_equal(y_pred.shape, y_test.shape)
```

Comment:
No, it breaks in `fit`. This seems a strange assertion. I'd appreciate it if you got rid of it. It might be worth noting that you're smoke testing this case to avoid a regression (and note the issue number).